pytest_html/plugin.py

```diff
@@ -215,21 +215,49 @@ def pytest_sessionfinish(self, session):
             html.title('Test Report'),
             html.style(raw(style_css)))
 
+        filter_checkbox = {}
+        test_result_type_amount = {'Passed': self.passed,
+                                   'Skipped': self.skipped,
+                                   'Failed': self.failed,
+                                   'Error': self.errors,
+                                   'XFailed': self.xfailed,
+                                   'XPassed': self.xpassed}
+
+        for test_type in test_result_type_amount:
+            if(test_result_type_amount[test_type] > 0):
+                oc_command = 'filter_table(' + test_type + ')'
+                filter_checkbox[test_type] = html.input(type='checkbox',
+                                                        checked='true',
+                                                        onChange=oc_command,
+                                                        name='filter_checkbox',
+                                                        id=test_type,
+                                                        value=test_type)
+            else:
+                filter_checkbox[test_type] = html.input(type='checkbox',
+                                                        disabled='true')
+
         summary = [html.h2('Summary'), html.p(
-            '{0} tests ran in {1:.2f} seconds.'.format(
+            '{0} tests ran in {1:.2f} seconds. '.format(
                 numtests, suite_time_delta),
+            html.p('(Un)check the boxes to filter the results.'),
             html.br(),
+            filter_checkbox['Passed'],
```

Comment:
The test file should be able to import from the plugin module, but I'm okay with creating a separate file if you prefer it. Rather than a dictionary, I wonder if a custom object might make more sense.